cookbooks/prescribed_velocity/prescribed_velocity.cc

```diff
@@ -110,6 +110,51 @@ namespace aspect
   }
 
   /**
+   * This function retrieves the unit support points (in the unit cell) for the current element
+   * The DGP elements used when 'set Use locally conservative discretization = true' is enabled do
+   * not have support points. If these elements are in use, a ficticious support point at the cell
+   * center is returned. This is necessary for the internal velocities to be constrained as desired.
+   */
+  template <int dim>
+  std::vector< Point<dim> >
+  get_unit_support_points_for_velocity(const SimulatorAccess<dim> &simulator_access,
+                                       const bool use_locally_conservative_discretization)
+  {
+    std::vector< Point<dim> > unit_support_points;
+    if ( !use_locally_conservative_discretization )
+      {
+        unit_support_points = simulator_access.get_fe().get_unit_support_points();
+      }
+    else
+      {
+        //special case for discontinuous pressure elements, which lack unit support points
+        const unsigned int dofs_per_cell = simulator_access.get_fe().dofs_per_cell;
+        for (unsigned int dof=0; dof < dofs_per_cell; dof++)
+          {
+            // base will hold element, base_index holds node/shape function within that element
+            const unsigned int
+            base       = simulator_access.get_fe().system_to_base_index(dof).first.first,
+            base_index = simulator_access.get_fe().system_to_base_index(dof).second;
+            // get the unit support points for the relevant element
+            std::vector< Point<dim> > my_support_points = simulator_access.get_fe().base_element(base).get_unit_support_points();
+            if ( my_support_points.size() == 0 )
+              {
+                //manufacture a support point, arbitrarily at cell center
+                if ( dim==2 )
+                  unit_support_points.push_back( Point<dim> (0.5,0.5) );
+                if ( dim==3 )
+                  unit_support_points.push_back( Point<dim> (0.5,0.5,0.5) );
```

Comment:
on second thought, rather than making the distinction between dimensions, just do
```
  // manufacture a support point, arbitrarily at cell's origin
  unit_support_points.push_back (Point<dim>());
```
This works independent of the dimension and is shorter.